src/lib/parser.rs

```diff
@@ -189,33 +189,28 @@ impl<'a, TokId: Clone + Copy + Debug + PartialEq + TryFrom<usize> + TryInto<usiz
         }
     }
 
-    /// Return a (`Lexeme`, `TIdx`) pair of the next lemexe. When `lexeme_prefix` is not
-    /// `None`, that is the next lexeme. Otherwise it will be the next lexeme in `self.lexemes` or
-    /// a specially constructed lexeme representing EOF.
-    pub(crate) fn next_lexeme(&self, lexemes_prefix: Option<Lexeme<TokId>>, la_idx: usize)
-                          -> (Lexeme<TokId>, TIdx)
+    /// Return a `Lexeme` for the next lemexe (if `la_idx` == `self.lexemes.len()` this will be
+    /// a lexeme constructed to look as if contains the EOF terminal).
+    pub(crate) fn next_lexeme(&self, la_idx: usize) -> Lexeme<TokId>
     {
-        if let Some(l) = lexemes_prefix {
-            (l, TIdx::from(l.tok_id().try_into().ok().unwrap()))
-        } else if let Some(l) = self.lexemes.get(la_idx) {
-            (*l, TIdx::from(l.tok_id().try_into().ok().unwrap()))
+        let ll = self.lexemes.len();
+        debug_assert!(la_idx <= ll);
```

Comment:
I wonder if `len` wouldn't be a slightly better variable name, since `ll` looks like `11` which gives the below if statement a completely different meaning. ;-)

Reply:
Point taken.